src/misc/ArborX_SymmetricSVD.hpp

```diff
@@ -77,133 +78,133 @@ KOKKOS_FUNCTION auto argmaxUpperTriangle(Matrix const &mat)
   return result;
 }
 
+// Compute x, y and theta such that
+// +---+---+              +---+---+
+// | a | b |              | x | 0 |
+// +---+---+ = R(theta) * +---+---+ * R(theta)^T
+// | b | c |              | 0 | y |
+// +---+---+              +---+---+
+template <typename Value>
+KOKKOS_FUNCTION void svd2x2(Value a, Value b, Value c, Value &x, Value &y,
+                            Value &cos_theta, Value &sin_theta)
+{
+  if (b == 0)
+  {
+    cos_theta = 1;
+    sin_theta = 0;
+    x = a;
+    y = c;
+    return;
+  }
+
+  // Calculate rotation angle theta ensuring |sin(theta)| <= |cos(theta)|, see
+  // https://en.wikipedia.org/wiki/Jacobi_rotation#Numerically_stable_computation
+  // or §8.4 in
+  // Golub, Gene H.; Van Loan, Charles F. (1996), Matrix Computations (3rd ed.),
+  // Baltimore: Johns Hopkins University Press, ISBN 978-0-8018-5414-9
+  auto tau = (c - a) / (2 * b);
+  auto tan_theta = Kokkos::copysign(
+      1 / (Kokkos::abs(tau) + Kokkos::sqrt(1 + tau * tau)), tau);
+
+  // FIXME: use rhypot
```

Comment:
Does that exist anywhere?
```suggestion
```

Reply:
In `Cuda` it does. Kokkos currently does not provide it.